src/main/java/net/javadiscord/javabot/events/SuggestionListener.java

```diff
@@ -68,21 +75,23 @@ private boolean canCreateSuggestion(MessageReceivedEvent event) {
 	}
 
 	/**
-	 * Adds all Attachments from the initial message to the new message action.
+	 * Adds all Attachments from the initial message to the new message action and sends the message.
 	 *
 	 * @param message The initial {@link Message} object.
 	 * @param action The new {@link MessageAction}.
-	 * @return The complete {@link MessageAction} with all attachments.
+	 * @return A {@link CompletableFuture} with the message that is being sent.
 	 */
-	private MessageAction addAttachments(Message message, MessageAction action) {
+	private CompletableFuture<Message> addAttachmentsAndSend(Message message, MessageAction action) {
```

Comment:
Moving the loop logic into a method, and using a stream to collect cleans this up a lot. 

`List<CompletableFuture<?>> attachmentFutures = message.getAttachments().stream()
		.map(attachment -> handleAttachment(action, attachment))
		.collect(Collectors.toList());`

Reply:
To be honest, moving this code to a method isn't worth it in my opinion since the code is pretty obvious.